src/scanner.c

```diff
@@ -382,6 +385,7 @@ UInt OpenInputStream (
     }
```

Comment:
The 'if' statement just above here may as well be removed (it's body should be kept, as it will now always be true. Perhaps move the assert above)?